src/python/pants/init/util.py

```diff
@@ -6,18 +6,19 @@
 from pants.fs.fs import safe_filename_from_path
 from pants.goal.goal import Goal
 from pants.init.options_initializer import BuildConfigInitializer
+from pants.option.option_value_container import OptionValueContainer
 from pants.subsystem.subsystem import Subsystem
 from pants.util.dirutil import absolute_symlink, safe_mkdir, safe_rmtree
 
 
-def init_workdir(global_options):
+def init_workdir(global_options: OptionValueContainer) -> str:
     """Given the bootstrap options (generally immediately after bootstrap), initialize the workdir.
 
     If it is in use, the "physical" workdir is a directory under the `pants_physical_workdir_base`
     that is unique to each working copy (via including the entire path to the working copy in its
     name using `safe_filename_from_path`).
     """
-    workdir_src = global_options.pants_workdir
+    workdir_src: str = global_options.pants_workdir
```

Comment:
Nit: preferred to use `cast()` instead. It better expresses the intent and MyPy will tell us when we can remove the `cast`.